client/layout/masterbar/masterbar.jsx

```diff
@@ -5,16 +5,23 @@ import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import React from 'react';
 
+/**
+ * Internal dependencies
+ */
+import { isWpMobileApp } from 'calypso/lib/mobile-app';
+
 /**
  * Style dependencies
  */
 import './style.scss';
 
-const Masterbar = ( { children, className } ) => (
-	<header id="header" className={ classNames( 'masterbar', className ) }>
-		{ children }
-	</header>
-);
+const Masterbar = ( { children, className } ) => {
+	return isWpMobileApp() ? null : (
```

Comment:
This doesn't seem like the right place for this logic - it is too late, and even if we're hiding the masterbar, all the code is loaded anyway. 

Have we considered doing it as early as possible? Ideally, we wouldn't have to load any of the chunks that are coming from components that are related to (embedded in) the masterbar at all.

Reply:
This should no be resoled I removed this changes.